site/app/models/gradeable/Gradeable.php

```diff
@@ -2593,4 +2513,419 @@ public function hasLeaderboard(): bool {
+        return $this->late_days;
+    }
+
+    public function isLateSubmissionAllowed(): bool {
+        return $this->late_submission_allowed;
+    }
+
+    public function setLateSubmissionAllowed(bool $allow_late_submission): void {
+        $this->late_submission_allowed = $allow_late_submission;
+        $this->modified = true;
+    }
+
+    public function getPrecision(): float {
+        return $this->precision;
+    }
+
+    /**
+     * Gets the components for this gradeable
+     * @return Component[]
+     */
+    public function getComponents(): array {
+        return $this->components;
+    }
+
+    public function setAllowedMinutes(?int $minutes): void {
+        $this->allowed_minutes = $minutes;
+        $this->modified = true;
+    }
+
+    public function getDependsOn(): ?string {
+        return $this->depends_on;
+    }
+
+    public function setDependsOn(?string $depends_on): void {
+        $this->depends_on = $depends_on;
+        $this->modified = true;
+    }
+
+    public function getDependsOnPoints(): ?int {
+        return $this->depends_on_points;
+    }
+
+    public function setDependsOnPoints(?int $depends_on_points): void {
+        $this->depends_on_points = $depends_on_points;
+        $this->modified = true;
+    }
+
+    public function setAnyManualGrades(bool $any_manual_grades): void {
+        $this->any_manual_grades = $any_manual_grades;
+        $this->modified = true;
+    }
+
+    public function isGradeInquiryAllowed(): bool {
+        return $this->grade_inquiry_allowed;
+    }
+
+    public function isGradeInquiryPerComponentAllowed(): bool {
+        return $this->grade_inquiry_per_component_allowed;
+    }
+
+    public function setGradeInquiryPerComponentAllowed(bool $is_grade_inquiry_per_component): void {
+        $this->grade_inquiry_per_component_allowed = $is_grade_inquiry_per_component;
+        $this->modified = true;
+    }
+
+    public function isDiscussionBased(): bool {
+        return $this->discussion_based;
+    }
+
+    public function setDiscussionBased(bool $discussion_based): void {
+        $this->discussion_based = $discussion_based;
+        $this->modified = true;
+    }
+
+    /**
+     * Gets the discussion thread id for this gradeable
+     * @return array<int>
+     */
+    public function getDiscussionThreadId(): array {
+        return $this->discussion_thread_id;
+    }
+
+    /**
+     * Sets the discussion thread id for this gradeable
+     * @param array<string>|array<int>|int|string $discussion_thread_id
+     */
+    public function setDiscussionThreadId($discussion_thread_id): void {
+        # If $discussion_thread_id is json encoded, decode it
+        if (is_string($discussion_thread_id) && is_array(json_decode($discussion_thread_id, true))) {
+            $discussion_thread_id = json_decode($discussion_thread_id, true);
+        }
+        $this->discussion_thread_id = array_map('intval', (array) $discussion_thread_id);
+        $this->modified = true;
```

Comment:
It looks like this setter is only used once, so you could probably just fix that instead of adding all of this complicated logic.